flink-table/flink-table-common/src/main/java/org/apache/flink/table/data/RowData.java

```diff
@@ -287,9 +287,6 @@ static FieldGetter createFieldGetter(LogicalType fieldType, int fieldPos) {
             default:
                 throw new IllegalArgumentException();
         }
-        if (!fieldType.isNullable()) {
```

Comment:
Can you add this to the JavaDoc of the Field getter interface?
// Converters and serializers always support nullability. The NOT NULL constraint is only considered on SQL
// semantic level but not data transfer. E.g. partial deletes (i.e. key-only upserts) set all non-key fields to
// null, regardless of logical type.